synoptic/src/synoptic/model/export/JsonExporter.java

```diff
@@ -96,151 +110,326 @@ public EventInstance(int traceID, int eventIndexWithinTrace) {
         }
     }
 
+    // Map of nodes and their respective global ID's
+    static Map<EvBasedNode, Integer> nodesIDMap = new LinkedHashMap<EvBasedNode, Integer>();
```

Comment:
Instead of static, make this private, and move it to the top of the class by globalID. Also, here and everywhere in this class where you instantiate classes with generics, use Java 7 style for brevity: `new LinkedHashMap<>();` Also also, to prevent bugs like the one I recall discussing in a video call where one of these global maps wasn't populated because you called the methods in an improper order, initially assign this to `null`, then construct the map within the method that populates this map, `makeNodesJSON`, then assert at the top of methods that *require* this map that it is *not* null (so assert at the top of `makeEdgesJSON`).